src/page/sinfo.rs

```diff
@@ -32,28 +32,39 @@ impl core::fmt::Display for SecInfo {
         match self.class {
             Class::Secs => write!(f, "S"),
             Class::Tcs => write!(f, "T"),
-            Class::Reg => write!(f, "{}", self.flags),
-            Class::Va => write!(f, "V"),
-            Class::Trim => write!(f, "^"),
+            Class::Regular => write!(f, "{}", self.flags),
+            Class::VersionArray => write!(f, "V"),
+            Class::Trimmed => write!(f, "^"),
+            Class::ShadowStackFirst => write!(f, "SSF"),
+            Class::ShadowStackRest => write!(f, "SSR"),
         }
     }
 }
 
-impl SecInfo {
-    /// Creates a `SecInfo` instance for regular pages
-    pub const fn reg(flags: Flags) -> Self {
+impl From<Class> for SecInfo {
+    fn from(class: Class) -> Self {
+        let flags = match class {
+            // A CPU constraint
+            Class::Tcs => Flags::READ | Flags::WRITE,
```

Comment:
I think the default here needs to be no flags. The reason is that the Linux kernel doesn't support any flags in the ioctl and will fail the request.

Reply:
Ah right, this is my fault :-) If I did that part now I would just check for RW. Actually I'd set empty for all. Because then someone could some day used by something else than Linux kernel. Let's not put policy here.

Reply:
Or actually for regular I will set 'R'. It's hardware constraint.